src/Prism.Plugin.Popups/PopupPageNavigationService.cs

```diff
@@ -300,7 +300,11 @@ protected override async Task DoPush(Page currentPage, Page page, bool? useModal
                     break;
                 default:
                     if (_popupNavigation.PopupStack.Any())
+                    {
                         await _popupNavigation.PopAllAsync(animated);
+                        currentPage = _applicationProvider.MainPage;
```

Comment:
This will only work for those scenarios where you have exactly one page in your Navigation Stack. Since this is used immediately after we've cleared the Popup Stack you'll want to call Prism's PageUtilities class here to get the current Page. 

```cs
currentPage = PageUtilities.GetCurrentPage(_applicationProvider.MainPage);
```

Reply:
Done.